rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -41,170 +41,284 @@ using Entity = liveliness::Entity;
 using EntityType = liveliness::EntityType;
 
 ///=============================================================================
-TopicStats::TopicStats(std::size_t pub_count, std::size_t sub_count)
-: pub_count_(pub_count),
-  sub_count_(sub_count)
+TopicData::TopicData(
+  liveliness::TopicInfo info,
+  std::unordered_set<liveliness::Entity> pubs,
+  std::unordered_set<liveliness::Entity> subs)
+: info_(std::move(info)),
+  pubs_(std::move(pubs)),
+  subs_(std::move(subs))
+{}
+
+///=============================================================================
+GraphCache::GraphCache(const z_id_t & zid)
+: zid_str_(liveliness::zid_to_str(zid))
 {
   // Do nothing.
 }
 
 ///=============================================================================
-TopicData::TopicData(
-  liveliness::TopicInfo info,
-  TopicStats stats)
-: info_(std::move(info)),
-  stats_(std::move(stats))
-{}
+std::shared_ptr<GraphNode> GraphCache::make_graph_node(const Entity & entity) const
```

Comment:
Should we just turn `GraphNode` into a class, and have a proper constructor for it?

Reply:
Could we save it for another ticket? I've opened https://github.com/ros2/rmw_zenoh/issues/132 to keep track.